### ISSUE TITLE
Inconsistent color naming between System.ConsoleColor and $PSStyle

### ISSUE THREAD

**Issue by hober** [PowerShell/PowerShell#16723]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

(Oversimplifying) `System.ConsoleColor` defines 8 "regular" colors and 8 corresponding "dark" colors, e.g. `Yellow` and `DarkYellow`.

The same 16 colors are represented in `$PSStyle` members as 8 "regular" colors and 8 corresponding "bright" colors, e.g. `Yellow` and `BrightYellow`.

`System.ConsoleColor`'s `DarkYellow` is `$PSStyle`'s `Yellow`, and `System.ConsoleColor`'s `Yellow` is `$PSStyle`'s `BrightYellow`.

This is confusing to say the least, and should probably be made consistent.

### Expected behavior

I would expect `System.ConsoleColor`'s `Yellow` and `$PSStyle`'s `Yellow` to be the same color.

### Actual behavior

`System.ConsoleColor`'s `DarkYellow` is `$PSStyle`'s `Yellow`, and `System.ConsoleColor`'s `Yellow` is `$PSStyle`'s `BrightYellow`.

### Environment data

```powershell
PSVersion                      7.2.1
PSEdition                      Core
GitCommitId                    7.2.1
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by jhoneill** [comment]
There are multiple annoyances here. 

1.  It seems the ANSI names are (e.g). Red and Bright Red, and that is the convention used in the settings for VS Code and Windows Terminal (the two main places I use PowerShell)  Back when someone decided what the names should be for `ConsoleColor` they picked Dark Red and Red , so PSStyle is "right" and what we have done up to now is "wrong". Making people who are familiar  with the ANSI  naming switch to the names in ConsoleColor is bad, making people familiar with ConsoleColor go the other way isn't good. 
2. We often want to specify a color parameter and help the user to complete it by giving the parameter an Enum type; if there were a new enum for names in `PSStyle` it we might not want to depend on it.  We can define our own enum or we can just use `ConsoleColor` (as used by `$host` and `Write-Host`) . We can't remove all doubt about whether  "Red" means the Lighter or Darker 
3.  The old style console host, Windows terminal, and VS Code all define what their idea of Red / light / dark red are.  Windows PowerShell redefined the console colours for dark yellow to ~ white and dark magenta to "PowerShell blue", so       
`$e = [char]27; "$e[35m Hello $e[0m"`  will be visible in Windows Terminal but lost if I run PowerShell.exe, that's extreme but   
`$e = [char]27; "$e[33m Hello $e[0m"` renders significantly differently in different places - in VSCode the default RGB values for the 16 colours make bright variants almost identical to the non-bright ones. 

I think the best solution might be to add Script methods (or Script Properties) to `ConsoleColor`  making it easier to go from a parameter declared `[ConsoleColor]$foreground`  to a string with the ANSI codes added. Since these would be added by loading a type.ps1xml file, they can be added in any version making dependencies less of a problem.  But not knowing what formatting as yellow will look like when someone else runs a script is a problem that will be with us for a while.

**Comment by jhoneill** [comment]
I wrote a `types.ps1XML` file to do what I described in the last paragraph above. It gets the values from `$pstyle`  so it will honour any customizations, but if `$pstyle` isn't present it uses the default ansi values so it will work with Windows PowerShell 5. 

https://gist.github.com/jhoneill/5912cd1ecac551fc3ada13df599f2f67 

Some examples. 
![image](https://user-images.githubusercontent.com/16209593/149124739-e00c8d81-59b6-4a4b-b3b4-41e034c9b83f.png)

The last example is meant to be something you'd do inside a function where `$newItemColor` is a `[consoleColor]` parameter with a default. 

I think we are stuck with this inconsistent naming, but this is a way to keep using `[ConsoleColor]`  and also play reasonably nicely with `$psStyle`.
